jdbc-v2/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java

```diff
@@ -358,30 +369,49 @@ public boolean isValid(int timeout) throws SQLException {
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+            this.defaultQuerySettings.setOption(name, value);
+        } catch (Exception e) {
+            throw new SQLClientInfoException("Failed to set client info.", Collections.singletonMap(name, ClientInfoStatus.REASON_UNKNOWN), e);
+        }
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+        } catch (SQLException e) {
+            throw new SQLClientInfoException("Failed to set client info.", new HashMap<>(), e);
+        }
+
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            setClientInfo(entry.getKey().toString(), entry.getValue().toString());
```

Comment:
ClientInfo is only what we can send to a server: 
```
The driver stores the value specified in a suitable location in the database.
```

and this one 

```
 The driver shall generate a warning if the client info name specified is not recognized by the driver.
```

Currently we may need to accept only 
```
    static final String PROP_APPLICATION_NAME = "ApplicationName";
``` 

I think this came from PostgreSQL JDBC driver - so we need to investigate more.

Reply:
btw, each property is set as individual query in the PSQL JDBC